specs/protocol/safer-safes.md

```diff
@@ -0,0 +1,507 @@
+
+### `clearLivenessModule`
+
+Removes the liveness module configuration by a previously enabled `safe`.
+
+- The contract MUST NOT be enabled as a module on the `safe`.
+- MUST erase the existing `liveness_challenge_period` and `fallback_owner` data related to the calling `safe`.
+- If a challenge exists, it MUST be cancelled, including emitting the appropriate events.
+- MUST emit a `ModuleCleared` event.
+
+### `configureTimelockGuard`
+
+Configure the contract as a timelock guard by setting the `timelock_delay`.
+
+- MUST allow an arbitrary number of `safe` contracts to use the contract as a guard.
+- The contract MUST be enabled as a guard for the `safe`.
+- MUST revert if `timelock_delay` is longer than 1 year.
+- MUST set the caller as a `safe`.
+- MUST take `timelock_delay` as a parameter and store is as related to the `safe`.
+- MUST emit a `GuardConfigured` event with at least `timelock_delay` as a parameter.
+
+### `clearTimelockGuard`
+
+Remove the timelock guard configuration by a previously enabled `safe`.
+
+- The contract MUST NOT be enabled as a guard for the `safe`.
+- MUST erase the existing `timelock_delay` data related to the calling `safe`.
+- If a challenge exists, it MUST be cancelled, including emitting the appropriate events.
+- MUST emit a `GuardCleared` event.
+
+### `viewLivenessModuleConfiguration`
+
+Returns the `liveness_challenge_period` and `fallback_owner` for a given `safe`.
+
+- MUST never revert.
+
+### `viewTimelockGuardConfiguration`
+
+Returns the `timelock_delay` for a given `safe`.
+
+- MUST never revert.
+
+### `getLivenessChallengePeriodEnd`
+
+Returns `challenge_start_time + liveness_challenge_period + timelock_delay` if there is a challenge for the given
+`safe`, or 0 if not.
+
+- MUST never revert.
+
+### `challenge`
+
+Challenges an enabled `safe` to prove that it is not in a liveness failure state.
+
+- MUST only be executable by `fallback` owner of the challenged `safe`.
+- MUST revert if the `safe` hasn't enabled the contract as a module.
+- MUST revert if the `safe` hasn't configured the contract as a module.
+- MUST revert if a challenge for the `safe` exists.
+- MUST set `challenge_start_time` to the current block time.
+- MUST emit the `ChallengeStarted` event.
+
+### `respond`
+
+Cancels a challenge for an enabled `safe`.
+
+- MUST only be executable by an enabled `safe`.
+- MUST revert if the `safe` hasn't enabled the contract as a module.
+- MUST revert if the `safe` hasn't configured the contract as a module.
+- MUST revert if there isn't a challenge for the calling `safe`.
+- MUST reset `challenge_start_time` to 0.
+- MUST emit the `ChallengeCancelled` event.
+
+### `changeOwnershipToFallback`
+
+With a successful challenge, removes all current owners from an enabled `safe`, appoints `fallback` as its sole owner,
+and sets its quorum to 1.
+
+- MUST be executable by anyone.
```

Comment:
It looks like smartcontracts disagrees here: https://github.com/ethereum-optimism/optimism/pull/17272#discussion_r2326104783